aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -478,6 +478,20 @@ module aptos_framework::coin {
         (option::extract(burn_ref_opt), BurnRefReceipt { metadata })
     }
 
+    // Permanently convert to BurnRef, and take it from the pairing.
+    // (i.e. future calls to borrow/convert BurnRef will fail)
+    public fun convert_and_take_paired_burn_ref<CoinType>(
```

Comment:
cc @lightmark. This might not make sense here before the end of the migration, as taking BurnRef means that the conversion for migration no longer works

Reply:
conversion doesn't use BurnRef, but uses fungible_asset::burn_internal

This prevents calling get_paired_burn_ref afterwards. I am not sure if we can clone BurnRef somehow, and return it and keep it in pairing as well.

Also I am not sure if caller can have multiple BurnCapability objects, that each needs to be converted.

@lightmark , thoughts?


For APT, we call this only after features::operations_default_to_fa_apt_store() flag has been set, which will not be set until there are no more APT CoinStores

Reply:
@lightmark does framework have multiple caps? If so - how are we going to convert them to Refs? do we need to actually clone/copy here with some friend function?

Reply:
We have two mint caps...lemme think about it.

Reply:
I think we cannot generate refs... mint cap should be stored in stake.move and burn is in transaction_fee.move ( transaction_fee is a friend of stake)

Reply:
ok cool, so for this PR we are good (it is burn ref we are touching).
for transforming mint ref, we need to consolidate transaction_fee and stake modules